core/services/llo/observation/data_source.go

```diff
@@ -354,21 +347,27 @@ func (d *dataSource) getObservableStreams() (llo.DSOpts, llo.StreamValues, time.
 	}
 	d.configDigestToStreamMu.Unlock()
 
-	// deduplicate streams and get the active ocr instance options
 	for _, vals := range streamsToObserve {
 		active, err := vals.IsActive()
-		if !active {
-			continue
-		}
-
 		if err != nil {
 			d.lggr.Errorw("getObservableStreams: failed to check if OCR instance is active", "error", err)
 			continue
 		}
 
-		return vals.opts, vals.streamValues, vals.observationInterval
+		if active {
+			return vals.opts, vals.streamValues, vals.observationInterval
+		}
+
 	}
 
 	d.lggr.Errorw("getObservableStreams: no active OCR instance found")
 	return nil, nil, 0
 }
+
+func closeIfOpen(ch chan struct{}) {
```

Comment:
Why is there uncertainty about whether the channel is open? This is typically a code smell leading to a race. The owner of the channel should be able to close once with confidence.

Reply:
We have a channel which indicates when the first run of the loop is complete and the caller waits on that channel. This channel is closed at the end of the first loop, if that loop is successful. On each subsequence loop, we check the input and might exit the loop if the input is invalid. This function allows us to not check if this is the first loop or a subsequent one.